src/pages/homepage.tsx

```diff
@@ -23,10 +23,15 @@ const HomePage = ({ mdxContent }: WithMdxResult) => {
   });
 
   const onSubmit: SearchProps['onSearchSubmit'] = async (url) => {
-    try {
-      router.push(`/service?${commonUrlParams}&url=${encodeURIComponent(url)}`);
-    } catch (e) {
-      console.error(e);
+    if ((url.toLowerCase().includes("localhost")) || (url.toLowerCase().includes("127.0.0.1"))) {
```

Comment:
This is a weak sanitisation, as other meta-addresses such as `0.0.0.0`, the whole `127.0.0.1/8` block or the IPv6 `::1` would all pass. This also does not prevent access to other IPs on the local network.